tooling/bundler/src/bundle/windows/sign.rs

```diff
@@ -88,93 +240,84 @@ fn locate_signtool() -> crate::Result<PathBuf> {
   Err(crate::Error::SignToolNotFound)
 }
 
-/// Check if binary is already signed.
-/// Used to skip sidecar binaries that are already signed.
-pub fn verify(path: &Path) -> crate::Result<bool> {
-  // Construct SignTool command
-  let signtool = locate_signtool()?;
-
-  let mut cmd = Command::new(&signtool);
-  cmd.arg("verify");
-  cmd.arg("/pa");
-  cmd.arg(path);
-
-  Ok(cmd.status()?.success())
-}
-
-pub fn sign_command(path: &str, params: &SignParams) -> crate::Result<(Command, PathBuf)> {
-  // Construct SignTool command
-  let signtool = locate_signtool()?;
-
-  let mut cmd = Command::new(&signtool);
-  cmd.arg("sign");
-  cmd.args(["/fd", &params.digest_algorithm]);
-  cmd.args(["/sha1", &params.certificate_thumbprint]);
-  cmd.args(["/d", &params.product_name]);
-
-  if let Some(ref timestamp_url) = params.timestamp_url {
-    if params.tsp {
-      cmd.args(["/tr", timestamp_url]);
-      cmd.args(["/td", &params.digest_algorithm]);
-    } else {
-      cmd.args(["/t", timestamp_url]);
-    }
-  }
-
-  cmd.arg(path);
-
-  Ok((cmd, signtool))
+/// This is check if azuresigntool is available on the users system.
+fn locate_azuresigntool() -> crate::Result<PathBuf> {
+  let mut cmd = Command::new("azuresigntool.exe");
+  cmd.arg("--help");
+  let _ = cmd.output_ok()?;
+  Ok(PathBuf::from("azuresigntool.exe"))
 }
 
-pub fn sign<P: AsRef<Path>>(path: P, params: &SignParams) -> crate::Result<()> {
-  let path_str = path.as_ref().to_str().unwrap();
-
-  info!(action = "Signing"; "{} with identity \"{}\"", path_str, params.certificate_thumbprint);
-
-  let (mut cmd, signtool) = sign_command(path_str, params)?;
-  debug!("Running signtool {:?}", signtool);
+impl Settings {
+  /// Attempts to create signing params from the environment variables and settings object.
+  /// If neither AzureSignTool or SignTool can be used because of missing environment variables, this will return `None`.
+  pub(crate) fn sign_params(&self) -> Option<SignParams> {
+    let product_name = self.product_name().to_string();
 
-  // Execute SignTool command
-  let output = cmd.output_ok()?;
+    // We'll attempt to create a SignParams struct from the environment variables and the settings
+    // First we'll start with AzureSignTool, if we have all the required environment variables set, then we'll use that.
+    // If not, we'll fallback to Signtool.
+    // But if we don't have a certificate thumbprint set, we can't sign at all and we'll return None.
+    match (
+      var_os("AZURE_KEYVAULT_URL"),
+      var_os("AZURE_CLIENT_ID"),
+      var_os("AZURE_TENANT_ID"),
+      var_os("AZURE_CLIENT_SECRET"),
+      var_os("AZURE_CERTIFICATE_NAME"),
```

Comment:
we need to list all of these new env variables in `tooling/cli/ENVIRONMENT_VARIABLES.md`

Reply:
Added!